model/src/pyrenew/deterministic/deterministic.py

```diff
@@ -63,10 +63,14 @@ def validate(value: ArrayLike) -> None:
         Raises
         ------
         Exception
-            If the input value object is not a ArrayLike.
+            If the input value object is not an ArrayLike object.
         """
         if not isinstance(value, ArrayLike):
-            raise Exception("value is not a ArrayLike")
+            raise ValueError(
+                f"value {value} passed to a DeterministicVariable "
+                f"is of type {type(value).__name__}, expected "
+                "an ArrayLike object"
+            )
```

Comment:
We have this type of check error throughout the project. Could be good to write a function for it.